### DISCUSSION TITLE
How to set conform.nvim format 4 spaces but not 2 spaces?

### DISCUSSION THREAD

**Discussion by taynpg** [LazyVim/LazyVim#2133] [Q&A] [answered]
# my options:
vim.o.tabstop = 4
vim.bo.tabstop = 4
vim.o.softtabstop = 4
vim.o.shiftround = true
vim.o.shiftwidth = 4
vim.bo.shiftwidth = 4

# Q: Now conform.nvim formats my cpp code to 2 indents, how do I set it to 4 Spaces? Thank you.

**Comment by madflow** [comment]
To my understanding - conform does not format your code. It redirects this task to a "formatter", that you have to configure. So if you use "prettier" or "formatterx", you will have to look into the configuration of that formatter.

**Reply by taynpg** [reply]
That's it, thank you very much for your help, let me know the direction of my efforts.

**Reply by dpetka2001** [reply, marked as answer]
Configure the `.clang-format` file at the root of your project according to the [docs](https://clang.llvm.org/docs/ClangFormatStyleOptions.html). There are many options there, so read through them to see what you want. My guess for your case would be the option `IndentWidth`, set that to 4.

**Reply by taynpg** [reply]
@dpetka2001 Your answer is correct, @madflow
Gave me directions, I just tried to succeed because I mistakenly wrote .clang-format to .clang_format, thank you very much for your help in configuring my own NVIM smoothly for a newbie like me.

**Reply by Raaghav-Arya** [reply]
Creating .clangd-format and adding the following:
```YAML
IndentWidth: 4
```
worked for me.

As mentioned in accepted answer.
